contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -2577,6 +2577,259 @@ AS 'babelfishpg_tsql', 'update_user_catalog_for_guest';
  
 CALL sys.babelfish_update_user_catalog_for_guest();
 
+ALTER VIEW sys.sp_sproc_columns_view RENAME TO sp_sproc_columns_view_deprecated_in_2_3_0;
+
+CREATE OR REPLACE VIEW sys.sp_sproc_columns_view
+AS
+-- Selects all parameters (input and output), but NOT return values
+SELECT
+    CAST(ss.name AS sys.sysname) AS PROCEDURE_QUALIFIER
+  , CAST(ss.orig_name AS sys.sysname) AS PROCEDURE_OWNER
+  , CAST(
+      CASE
+        WHEN ss.prokind = 'p' THEN CONCAT(ss.proname, ';1')
+        ELSE CONCAT(ss.proname, ';0')
+      END
+    AS sys.nvarchar(134)) AS PROCEDURE_NAME
+  , CAST(COALESCE(ss.proargnames[(ss.x).n], '') AS sys.SYSNAME) AS COLUMN_NAME
+  , CAST(
+ 	  CASE
+ 	  	WHEN ss.proargmodes[(ss.x).n] in ('o', 'b') THEN 2
+ 	  	ELSE 1
+ 	  END
+  	AS smallint) AS COLUMN_TYPE
+  , CAST(
+  	  CASE
+  	    WHEN st.is_table_type = 1 THEN -153
+  	    ELSE sdit.data_type 
+  	  END
+  	AS smallint) AS DATA_TYPE
+  , CAST(st.name AS sys.sysname) AS TYPE_NAME
+  , CAST(
+  	  CASE
+  	    WHEN st.is_table_type = 1 THEN 0
+  	    ELSE st.precision 
+  	  END 
+  	AS sys.int) AS PRECISION
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN 2147483647
+  	    ELSE st.max_length 
+  	  END
+  	AS sys.int) AS LENGTH
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN NULL
+  	    ELSE st.scale
+  	  END
+  	AS smallint) AS SCALE
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN NULL
+  	    ELSE sdit.num_prec_radix
+      END
+  	AS smallint) AS RADIX
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN 1
+  	    ELSE sdit.nullable 
+  	  END
+  	AS smallint) AS NULLABLE
+  , CAST(NULL AS sys.varchar(254)) AS REMARKS
+  , CAST(NULL AS sys.nvarchar(4000)) AS COLUMN_DEF
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN -153
+  	    ELSE sdit.sql_data_type 
+  	  END
+  	AS smallint) AS SQL_DATA_TYPE
+  , CAST(sdit.sql_datetime_sub AS smallint) AS SQL_DATETIME_SUB
+  , CAST(
+      CASE
+  	    WHEN st.is_table_type = 1 THEN 2147483647
+  	    ELSE NULL
+  	  END
+  	AS int) AS CHAR_OCTET_LENGTH
+  , CAST((ss.x).n AS INT) AS ORDINAL_POSITION
+  , CAST(
+      CASE
+        WHEN st.is_table_type = 1 THEN 'YES'
+        WHEN sdit.nullable = 1 THEN 'YES'
+        ELSE 'NO'
+      END
+    AS sys.varchar(254)) AS IS_NULLABLE
+  , CAST(
+  	  CASE
+  	    WHEN st.is_table_type = 1 THEN 0
+  	    ELSE sdit.ss_data_type
+  	  END
+  	AS sys.tinyint) AS SS_DATA_TYPE
+  , CAST(ss.proname AS sys.sysname) AS original_procedure_name
+  FROM pg_type t
+  INNER JOIN sys.types st ON st.user_type_id = t.oid
+  LEFT JOIN sys.spt_datatype_info_table sdit ON sdit.type_name = (SELECT name FROM sys.types WHERE user_type_id = st.system_type_id)
```

Comment:
Could you please elaborate more on the subquery? 
why join condition is not sdit.type_name = st.name

Reply:
Yes, so we need to join that to get the ODBC data type information for user-defined types. Currently, spt_datatype_info_table only have the primitive/system-defined types and are all hardcoded. Doing `sdit.type_name = st.name` would not work since spt_datatype_info_table doesn't track user-defined types since it isn't recorded so I did this condition so we can find what type the user type was based on and use that information.

Reply:
Added comments to the code to explain it as well